rcl/test/rcl/test_security.cpp

```diff
@@ -0,0 +1,221 @@
+    });
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      allocator.deallocate(secure_root, allocator.state);
+    });
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      allocator.deallocate(base_lookup_dir_fqn, allocator.state);
+    });
+  }
+
+  void set_base_lookup_dir_fqn(const char * resource_dir, const char * resource_dir_name)
+  {
+    base_lookup_dir_fqn = rcutils_join_path(
+      resource_dir, resource_dir_name, allocator);
+    std::string putenv_input = ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME "=";
+    putenv_input += base_lookup_dir_fqn;
+    memcpy(
+      g_envstring, putenv_input.c_str(),
+      std::min(putenv_input.length(), sizeof(g_envstring) - 1));
+    putenv_wrapper(g_envstring);
+  }
+
+  rcl_allocator_t allocator;
+  char * root_path;
+  char * secure_root;
+  char * base_lookup_dir_fqn;
+};
+
+TEST_F(TestGetSecureRoot, failureScenarios) {
+  ASSERT_EQ(
+    rcl_get_secure_root(TEST_SECURITY_CONTEXT_ABSOLUTE, &allocator),
+    (char *) NULL);
+  rcl_reset_error();
+
+  putenv_wrapper(ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME "=" TEST_RESOURCES_DIRECTORY);
+
+  /* Security directory is set, but there's no matching directory */
+  /// Wrong security context
+  ASSERT_EQ(
+    rcl_get_secure_root("some_other_security_context", &allocator),
+    (char *) NULL);
+  rcl_reset_error();
+}
+
+TEST_F(TestGetSecureRoot, successScenarios_local_exactMatch) {
+  putenv_wrapper(
+    ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME "="
+    TEST_RESOURCES_DIRECTORY TEST_SECURITY_DIRECTORY_RESOURCES_DIR_NAME);
+
+  secure_root = rcl_get_secure_root(TEST_SECURITY_CONTEXT_ABSOLUTE, &allocator);
+  std::string secure_root_str(secure_root);
+  ASSERT_STREQ(
+    TEST_SECURITY_CONTEXT,
+    secure_root_str.substr(secure_root_str.size() - (sizeof(TEST_SECURITY_CONTEXT) - 1)).c_str());
+}
+
+TEST_F(TestGetSecureRoot, successScenarios_local_exactMatch_multipleTokensName) {
+  putenv_wrapper(ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME "=" TEST_RESOURCES_DIRECTORY);
+
+  secure_root = rcl_get_secure_root(
+    TEST_SECURITY_DIRECTORY_RESOURCES_DIR_NAME PATH_SEPARATOR TEST_SECURITY_CONTEXT, &allocator);
+  std::string secure_root_str(secure_root);
+  ASSERT_STREQ(
+    TEST_SECURITY_CONTEXT,
+    secure_root_str.substr(secure_root_str.size() - (sizeof(TEST_SECURITY_CONTEXT) - 1)).c_str());
+}
+
+TEST_F(TestGetSecureRoot, nodeSecurityDirectoryOverride_validDirectory) {
+  /* Specify a valid directory */
+  putenv_wrapper(ROS_SECURITY_DIRECTORY_OVERRIDE "=" TEST_RESOURCES_DIRECTORY);
+  root_path = rcl_get_secure_root(
+    "name shouldn't matter", &allocator);
+  ASSERT_STREQ(root_path, TEST_RESOURCES_DIRECTORY);
+}
+
+TEST_F(
+  TestGetSecureRoot,
+  nodeSecurityDirectoryOverride_validDirectory_overrideRootDirectoryAttempt) {
+  /* Setting root dir has no effect */
+  putenv_wrapper(ROS_SECURITY_DIRECTORY_OVERRIDE "=" TEST_RESOURCES_DIRECTORY);
+  root_path = rcl_get_secure_root(
+    "name shouldn't matter", &allocator);
```

Comment:
@ivanpauno nit: do we need to wrap these arguments?

Reply:
See https://github.com/ros2/rcl/pull/515/commits/e3d5ebdf6094274f7f74d752ecce3fbadec84150.